src/app/reporting/Engine.h

```diff
@@ -119,6 +121,8 @@ class Engine
      *
      */
     uint32_t mCurReadHandlerIdx = 0;
+
+    ClusterInfo mChangeSetPathList[CHIP_IM_MAX_NUM_SERVER_DIRTY_PATHS];
```

Comment:
Updated code

Reply:
Updated code: ClusterInfo itself has mpNext inside, and there is one clusterInfo pool from interaction model engine, which currently holds the interested path for each active subscribe client, I am extending its responsibility and reuse its functionality to hold changed path as well during post-subscription, every time application call setChange/SetDirty, which would use the existing MergeOverlappedAttributePath and merge overlapped path into mpChangePathList, otherwise, it uses the existing PushFront to push one available clusterInfo into mpChangePathList, regarding MergeOverlappedAttributePath, this is one existing function in readhandler, i will relocate this function to interactionModel engine